scripts/test/system.sh

```diff
@@ -198,6 +206,13 @@ _subctl benchmark throughput "${KUBECONFIGS_DIR}"/kind-config-cluster1 "${KUBECO
 
 # Test subctl cloud prepare invocations
 
+# test non deploy subctl for deploy commands
+_non_deploy_subctl deploy-broker --kubeconfig "${KUBECONFIGS_DIR}"/kind-config-cluster1 && exit 1
+_non_deploy_subctl join --kubeconfig "${KUBECONFIGS_DIR}"/kind-config-cluster1 broker-info.subm --clusterid cluster1 && exit 1
```

Comment:
Add similar tests for all ignored commands. That way, we don't miss any. In future, if the build constraint is accidentally removed from some commands, this job should fail.